tests/codebuild/delete_tests.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+source common.sh
+source create_tests.sh
+
+# Verify that k8s and SageMaker resources are deleted when the user uses the delete verb.
+function run_delete_canary_tests
+{
+  verify_delete TrainingJob testfiles/xgboost-mnist-trainingjob.yaml
+
+  verify_delete HyperparameterTuningJob testfiles/xgboost-mnist-hpo.yaml
+
+  # Create model before running batch delete test
+  run_test testfiles/xgboost-model.yaml
+  verify_test Model xgboost-model 1m Created
+  yq w -i testfiles/xgboost-mnist-batchtransform.yaml "spec.modelName" "$(get_sagemaker_model_from_k8s_model xgboost-model)"
+
+  verify_delete BatchTransformJob testfiles/xgboost-mnist-batchtransform.yaml
+}
+
+function run_delete_integration_tests
+{
+  run_delete_canary_tests
+}
+
+# Applies a k8s resource, waits for it to start and then immediately deletes
+# and ensure the state of the resource in k8s is stopping.
+# Parameter:
+#    $1: CRD type
+#    $2: Filename of test
+#    $3: (Optional) Timeout
+# e.g. verify_delete trainingjob xgboost-mnist.yaml ".status.sageMakerTrainingJobName"
```

Comment:
I looked into case insensitive string matching in bash and it's basically a nightmare unless we do blocks of if-else statements rather than a switch. 
I'll update the example and add a comment about proper casing.

Reply:
To do case-insensitive matching, I'd do this:
```bash
local crd_type="$(echo "$1" | awk '{print tolower($0)}')"
```